src/metrics/metric/id/m3/id.go

```diff
@@ -165,7 +156,12 @@ func NewSortedTagIterator(sortedTagPairs []byte) id.SortedTagIterator {
 
 // NewPooledSortedTagIterator creates a new pooled sorted tag iterator.
 func NewPooledSortedTagIterator(sortedTagPairs []byte, pool id.SortedTagIteratorPool) id.SortedTagIterator {
-	it := &sortedTagIterator{pool: pool}
+	it := newPooledSortedTagIterator(sortedTagPairs, pool)
+	return &it
+}
+
+func newPooledSortedTagIterator(sortedTagPairs []byte, pool id.SortedTagIteratorPool) sortedTagIterator {
+	it := sortedTagIterator{pool: pool}
```

Comment:
shouldn't this do `pool.Get()`?

Reply:
I think that would be circular--this is *called* by `pool.Get` as part of the initialization flow. That is:

```
pool.Get() {
   if no objects present{
      // alloc
     NewPooledSortedTagIterator(nil, pool)
   } 
}

```

Reply:
talked offline: fine as is